### ISSUE TITLE
Accessibility Issue with Popover Trigger State Attributes

### ISSUE THREAD

**Issue by armchair-traveller** [melt-ui/melt-ui#1181]
### Describe the bug

The attributes `aria-expanded="true"` and `data-state="open"` remain set even when the popover is closed. These attributes should reflect the actual state of the popover. The corresponding issue does not affect the dialog component, as their trigger element updates attributes according to the open/closed state.

### Expected Behavior
When the popover is closed, `aria-expanded` should be set to `false` and `data-state` should be set to `closed`. This behavior ensures accurate accessibility and expected styles based on the `data-state` attribute.

### Affected Components
`melt-ui` popover trigger
`bits-ui` popover trigger
`shadcn-svelte` popover trigger

### Additional Context
This inconsistency in the attribute values may affect the accessibility of components and cause confusion for assistive technology users. Correcting this issue is crucial to ensure compliance with accessibility guidelines and improve the user experience for everyone.

### Reproduction

Check attributes of trigger in [our docs](https://melt-ui.com/docs/builders/popover)
Respectively, check [expected behavior](https://ui.shadcn.com/docs/components/popover)

### System Info

```bash
N/A
```

### Severity

annoyance
